app/src/main/java/com/internship/move/ui/authentication/register/DrivingLicenseFragment.kt

```diff
@@ -0,0 +1,67 @@
+package com.internship.move.ui.authentication.register
+
+import android.net.Uri
+import android.os.Bundle
+import android.view.View
+import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.content.FileProvider
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.fragment.findNavController
+import com.internship.move.BuildConfig
+import com.internship.move.R
+import com.internship.move.databinding.FragmentDrivingLicenseBinding
+import com.zhuinden.fragmentviewbindingdelegatekt.viewBinding
+import java.io.File
+
+class DrivingLicenseFragment : Fragment(R.layout.fragment_driving_license) {
+
+    private val binding by viewBinding(FragmentDrivingLicenseBinding::bind)
+    private val takeImageResult = registerForActivityResult(ActivityResultContracts.TakePicture()) { isSuccess ->
+        if (isSuccess) {
+            latestTmpUri?.let { uri ->
+                findNavController().navigate(DrivingLicenseFragmentDirections.actionDrivingLicenseFragmentToViewLicenseFragment(uri.toString()))
+            }
+        } else {
+            Toast.makeText(requireContext(), getString(R.string.driving_license_fail_message), Toast.LENGTH_SHORT).show()
+        }
+    }
+    private var latestTmpUri: Uri? = null
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        initToolbar()
+        binding.addLicenseBtn.setOnClickListener {
+            takePicture()
+        }
+    }
+
+    private fun initToolbar() {
+        val customToolbar = binding.toolbar
```

Comment:
I've left a comment regarding your implementation of the custom toolbar.
If you decide to change it as mentioned, and use `MaterialToolber` directly in the layouts that it's needed in, you can set most of these attributes in the xml file, except for the click listener

Also, if you change this you'll likely call `binding.toolbar.setNavigationOnClickListener` directly since it would be the only thing to set. But I'd like to give a couple suggestions about what you did here, for the future:
Most of the things we do are "custom" so this prefix can apply to anything, anywhere and becomes redundant. I would suggest either just calling it `toolbar` since it's only one in this Fragment and it doesn't need an intricate name. Or simply call `binding.toolbar` for each of the calls, imo it's not that big of an eyesore that some people make it out to be, but that's very subjective.